src/core/runner.ts

```diff
@@ -24,24 +39,75 @@ export async function run(rawOptions: RawOptions) {
     setOptions(rawOptions);
     let options = getOptions();
 
+    let partialCheck = await getPartialCheck();
+    if (options.partialCheckLimit) {
+        if (!partialCheck) {
+            reportWarning(
+                `Skipping fence validation. Could not calculate a partial check, but partialCheckLimit was specified in options`
+            );
+            return getResult();
+        }
+        if (
+            partialCheck.fences.length + partialCheck.sourceFiles.length >
+            options.partialCheckLimit
+        ) {
+            reportWarning(
+                `Skipping fence validation. The partial check had more than ${options.partialCheckLimit} changes`
+            );
+            return getResult();
+        }
+    }
+
     let sourceFileProvider: SourceFileProvider = options.looseRootFileDiscovery
         ? new FDirSourceFileProvider(options.project, options.rootDir)
         : new TypeScriptProgram(options.project);
 
-    // Do some sanity checks on the fences
-    validateTagsExist();
+    if (!partialCheck) {
+        // validating tags exist requires a full load of all fences
+        // we can't do this in partial check mode.
+        //
+        // Prefetching the full config set here avoids the overhead
+        // of partial fence loading, since we know we are loading
+        // the full fence set.
+        getConfigManager().getAllConfigs();
+        validateTagsExist();
+    } else {
+        reportWarning(
+            `skipping validateTagsExist. Cannot validate tag existence during partial checks`
```

Comment:
I'm torn on this particular warning. I'm of the opinion it makes sense to warn users when checks are being skipped, but these checks are being skipped any time a partial check is performed (which in the context of using this as a commit hook, means every time you commit).

I'm leaning towards removing the warning either in good-fences or by filtering it out in owa-build. Interested in hearing your opinions on the awkward ergonomics here @smikula

Reply:
I'm fine with removing it here.  This is more of a nice-to-have sanity check but doesn't affect the correctness of the validation.